ID:FPCO-25735;HOURS:2;DONE:90; Add timer in login command

ManishPrajapati-GoFynd opened the pull request.

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Add timer in login command

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Remove testing fields

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Set timer to 2 min | Add debug logs | Update alpha version

jinalvirani commented on src/lib/Auth.ts:
```diff
@@ -11,6 +10,9 @@ import chalk from 'chalk';
 import { getLocalBaseUrl } from '../helper/serve.utils';
+import Debug from './Debug';
+
+const SERVER_TIMER = 60000 * 2;
```
Change this to 5 min as per task description

ManishPrajapati-GoFynd replied:
This is resolved

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Set timer to 5 min

vivek-gofynd commented on src/lib/Auth.ts:
```diff
@@ -11,6 +10,9 @@ import chalk from 'chalk';
 import { getLocalBaseUrl } from '../helper/serve.utils';
+import Debug from './Debug';
+
+const SERVER_TIMER = 60000 * 5;
```
@manishprajapati52 keep it to 1 minute only. @manishprajapati52

ManishPrajapati-GoFynd replied:
@vivek-gofynd 
One minute would be less if user is not already signed in on partners panel. What do you think?

ManishPrajapati-GoFynd replied:
Set 2 minutes

vivek-gofynd commented on src/lib/Auth.ts:
```diff
@@ -50,13 +52,32 @@ export const getApp = async () => {
+function startTimer(){
+    Debug("Server timer starts")
+    Auth.timer_id = setTimeout(() => {
+        console.log(chalk.magenta('Server timeout: Please run fdk login command again.'));
```
Don't use console.log here @manishprajapati52

ManishPrajapati-GoFynd replied:
@vivek-gofynd 
This is debug log. I have added for debug purpose(debug.log file). Should I remove this?

ManishPrajapati-GoFynd replied:
Added in Debug log too...

vivek-gofynd commented on src/lib/Auth.ts:
```diff
@@ -50,13 +52,32 @@ export const getApp = async () => {
     Auth.server = serverIn.listen(port, (err) => {
-        if (err) console.log(err);
+        if (err) {
+            console.log(err);
```
Log error in debug file too. @manishprajapati52

ManishPrajapati-GoFynd replied:
Done

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Add debug logs and set 2 minute timeout

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Throw error on timeout

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Merge branch 'master'

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Merge branch 'master'

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; remove temp theme

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; add temp theme folder in git ignore

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Add partners domain support in env set command

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Fix support for sandbox

ManishPrajapati-GoFynd committed:
5.0.5

Commit:
Updated message

Commit:
Merge branch 'FPCO-25735-timeout-in-fdk-login' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Fix error handling of timeout | Fix error handling of port in use | improve user experience

ManishPrajapati-GoFynd committed:
Merge branch 'FPCO-25735-timeout-in-fdk-login' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;HOURS:2;DONE:90; Update error message and sandbox handling

Commit:
Error message updated

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; Remove auth command

ManishPrajapati-GoFynd committed:
Merge branch 'master' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; Fix test cases

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; Update readme

vivek-gofynd commented on src/commands/auth.ts:
```diff
@@ -6,6 +6,7 @@ export default function context(program: Command) {
     program
         .command('auth')
         .alias('login')
+        .option('-ad, --api-domain [partners-domain]', 'Partners domain')
```
"It will be Platform API domain" instead of "Partners domain" @manishprajapati52

ManishPrajapati-GoFynd replied:
Updated✅

vivek-gofynd commented on src/fdk.ts:
```diff
@@ -241,14 +246,14 @@ export async function init(programName: string) {
         console.warn(
             chalk.yellow(
-                `Warning: Reseting active environment to api.fynd.com. Please use \`fdk env set -u <env-api-url>\` to change active environment. Ref: ${
+                `Warning: Reseting active environment to api.fynd.com. Please use \`fdk env set -p <partners-domain>\` to change active environment. Ref: ${
```
Need to update this according to new changes

ManishPrajapati-GoFynd replied:
Updated✅

vivek-gofynd replied:
We are removing fdk env set command itslef here @manishprajapati52

ManishPrajapati-GoFynd replied:
This is updated.

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; update warning message

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; Fix option description

ManishPrajapati-GoFynd committed:
ID:FPCO-25735;DONE:100;HOURS:4; Fix warning message

ManishPrajapati-GoFynd committed:
Merge branch 'master' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

bhargavprajapatiFynd committed:
Improved extension preview-url command user experience

bhargavprajapatiFynd committed:
Updated ngrok tunnel related logs

bhargavprajapatiFynd committed:
improved extension preview-url command

bhargavprajapatiFynd committed:
minor improvements

jinalviranii committed:
Add ext list and update fdk.ext.config read logic

Commit:
Merge branch 'master' of github.com:gofynd/fdk-cli into improve-extension-preview-url-command

Commit:
branches added

Commit:
ngork tunnel support removed

ManishPrajapati-GoFynd committed:
handle submodule in extension template

ManishPrajapati-GoFynd committed:
handle git command error

ManishPrajapati-GoFynd committed:
add debug logs

Commit:
preview URL changes

ManishPrajapati-GoFynd committed:
create action in temporary folder and then move it to extension folder if succeed

ManishPrajapati-GoFynd committed:
move files using cross OS compatible commands

ManishPrajapati-GoFynd committed:
move function to utils with doc string comments | create branch variable

ManishPrajapati-GoFynd committed:
fix install java frontend dependencies

bhargavprajapatiFynd committed:
added --auto-update and --tunnel-url flag support and improved error handling

bhargavprajapatiFynd committed:
removed dead code

ManishPrajapati-GoFynd committed:
Update frontend port and add .env in java extension too

ManishPrajapati-GoFynd committed:
Update test case

ManishPrajapati-GoFynd committed:
Update frontend port to 8081 as env path is updated in frontend

ManishPrajapati-GoFynd committed:
Update base url in both env and application.yml if exist

ManishPrajapati-GoFynd committed:
add temp folder constant and stringify error

bhargavprajapatiFynd committed:
Merge branch 'handle-submodule-in-extension-template--master' into improve-extension-preview-url-command

bhargavprajapatiFynd committed:
Added extension config file support

bhargavprajapatiFynd committed:
changed extension context file name

bhargavprajapatiFynd committed:
Fixed extension preview url test cases

bhargavprajapatiFynd committed:
Remove updating env in extension init and setup command

bhargavprajapatiFynd committed:
Updated testcases for .env file change

bhargavprajapatiFynd committed:
Adding extension.context file on ext init and setup

bhargavprajapatiFynd committed:
fix: added proper api secret in context file

bhargavprajapatiFynd committed:
Added new ext tunnel command

Commit:
Readme.md updated

Commit:
6.0.0

Commit:
6.0.0-beta.0

Commit:
6.0.0-beta.1

Commit:
added missing dependency

Commit:
6.0.0-beta.2

bhargavprajapatiFynd committed:
Made node+react as first option in ext init and setup command

ManishPrajapati-GoFynd committed:
add template support in init extension command

bhargavprajapatiFynd committed:
- Added --reset flag
- improved code readability by refactoring config to context for extension context
- changed --auto-update flag to --no-auto-update flag to make it inline with shell command standards
- Updated readme.md for new changes

ManishPrajapati-GoFynd committed:
Ask developer to select or create extension on init command

ManishPrajapati-GoFynd committed:
Enhancement on init command

bhargavprajapatiFynd committed:
Added sticky preview url

ManishPrajapati-GoFynd committed:
Merge branch 'improve-extension-preview-url-command' of github.com:gofynd/fdk-cli into feature--improve-extension-preview-url-command

ManishPrajapati-GoFynd committed:
Fix test cases

ManishPrajapati-GoFynd committed:
Fix mock argument

ManishPrajapati-GoFynd committed:
Merge branch 'master' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

ManishPrajapati-GoFynd committed:
Merge branch 'improve-extension-preview-url-command' of github.com:gofynd/fdk-cli into FPCO-25735-timeout-in-fdk-login

ManishPrajapati-GoFynd committed:
Updating messages and removing extra lines

ManishPrajapati-GoFynd committed:
Update test case

bhargavprajapatiFynd committed:
revert timer changes

bhargavprajapatiFynd committed:
Merge branch 'master' into FPCO-25735-timeout-in-fdk-login

bhargavprajapatiFynd committed:
Merge branch 'master' into FPCO-25735-timeout-in-fdk-login

bhargavprajapatiFynd committed:
only run auth file
